packages/fetch/src/fetch.ts

```diff
@@ -0,0 +1,244 @@
+  >
+): FetchReturn<Result, undefined>;
+export function createFetch<Result, InitialValue extends Result | undefined = undefined>(
+  ...args: DistributeFetcherArgs<
+    FetchArgs,
+    [options: FetchOptions<Result, InitialValue, FetchArgs>]
+  >
+): FetchReturn<Result, InitialValue>;
+export function createFetch<Result, InitialValue extends Result | undefined = undefined>(
+  ...args: DistributeFetcherArgs<
+    FetchArgs,
+    [
+      options: FetchOptions<Result, InitialValue, FetchArgs>,
+      modifiers: ReturnType<RequestModifier>[]
+    ]
+  >
+): FetchReturn<Result, InitialValue>;
+export function createFetch<
+  Result,
+  InitialValue extends Result | undefined = undefined,
+  FetcherArgs extends any[] = FetchArgs
+>(
+  ...args: DistributeFetcherArgs<
+    FetcherArgs,
+    [options: FetchOptions<Result, InitialValue, FetcherArgs>]
+  >
+): FetchReturn<Result, InitialValue>;
+export function createFetch<
+  Result,
+  InitialValue extends Result | undefined = undefined,
+  FetcherArgs extends any[] = FetchArgs
+>(
+  ...args: DistributeFetcherArgs<
+    FetcherArgs,
+    [
+      options: FetchOptions<Result, InitialValue, FetcherArgs>,
+      modifiers: ReturnType<RequestModifier>[]
+    ]
+  >
+): FetchReturn<Result, InitialValue>;
+export function createFetch<
+  Result,
+  InitialValue extends Result | undefined,
+  FetcherArgs extends any[]
+>(
+  ...args:
+    | DistributeFetcherArgs<FetchArgs, []>
+    | DistributeFetcherArgs<FetchArgs, [options: FetchOptions<Result, InitialValue, FetchArgs>]>
+    | DistributeFetcherArgs<FetchArgs, [modifiers: ReturnType<RequestModifier>[]]>
+    | DistributeFetcherArgs<
+        FetchArgs,
+        [
+          options: FetchOptions<Result, InitialValue, FetchArgs>,
+          modifiers: ReturnType<RequestModifier>[]
+        ]
+      >
+    | DistributeFetcherArgs<FetcherArgs, [options: FetchOptions<Result, InitialValue, FetcherArgs>]>
+    | DistributeFetcherArgs<FetcherArgs, [modifiers: ReturnType<RequestModifier>[]]>
+    | DistributeFetcherArgs<
+        FetcherArgs,
+        [
+          options: FetchOptions<Result, InitialValue, FetcherArgs>,
+          modifiers: ReturnType<RequestModifier>[]
+        ]
+      >
+): FetchReturn<Result, InitialValue> {
+  const options = ([args[2], args[1]].find(isOptions) || {}) as FetchOptions<
+    Result,
+    InitialValue,
+    FetcherArgs
+  >;
+  const urlAccessor: Accessor<FetcherArgs | undefined> = createMemo(() => {
+    if (options.disable) {
+      return undefined;
+    }
+    const info: FetcherArgs[0] | undefined =
+      typeof args[0] === "function"
+        ? (args[0] as Accessor<FetcherArgs | FetcherArgs[0]>)()
+        : args[0];
+    if (!info) {
+      return undefined;
+    }
+    const init =
+      typeof args[1] === "function"
+        ? (args[1] as Accessor<FetcherArgs[1]>)()
+        : isOptions(args[1])
+        ? undefined
+        : (args[1] as RequestInit);
```

Comment:
While I wasn't sure this would make a difference since in this case the memo would be static so it would basically only use a little more memory, it's still something that can be optimized. I'll rewrite this part.